library/ssl_cli.c

```diff
@@ -3336,7 +3337,11 @@ static int ssl_write_client_key_exchange( mbedtls_ssl_context *ssl )
                                         own_pubkey, own_pubkey_max_len,
                                         &own_pubkey_len );
         if( status != PSA_SUCCESS )
+        {
+            if( psa_destroy_key( handshake->ecdh_psa_privkey ) == PSA_SUCCESS )
+                handshake->ecdh_psa_privkey = MBEDTLS_SVC_KEY_ID_INIT;
```

Comment:
Not sure if the check hewre is really needed, I'd set `handshake->ecdh_psa_privkey = MBEDTLS_SVC_KEY_ID_INIT` whatever thew result of `psa_destroy_key`

Reply:
I assumed that it should be set only in case of a succesfull destruction, as was the behaviour before.

Reply:
I do not know the current policy about that, setting the key id to `MBEDTLS_SVC_KEY_ID_INIT` after a key destroy isn't used a lot in the library